### ISSUE TITLE
[Feature request] Interface for attribute definitions

### ISSUE THREAD

**Issue by mm21** [mm21/trilium-alchemy#3]
Spinning off discussion from https://github.com/zadam/trilium/discussions/4144#discussioncomment-6681844.

Currently you can use `len(note.attributes["myLabel"])` to get the number of attributes by name, but that doesn't tell you if it's multi- or single-valued according to the attribute definition.

One quick way to check if an attribute is single- or multi-valued would be checking e.g. `"single" in note["label:myLabel"]`, but it would be better if the values were parsed out for you. I had an idea that would look something like:

```python
my_label_def = note.attributes.defs["myLabel"] # get attribute with name label:myLabel

if my_label_def.single:
    ...
else:
    ...
```

Basically `Note.attributes.defs` would function similarly to `Note.attributes`, but it would prepend `label:` or `relation:` to the name, and you'd get a `LabelDef` - a subclass of `Label` which additionally has `promoted`, `multi`, and `value_type`. `RelationDef` would have `promoted`, `multi`, and `inverse`. These `Label` subclasses would simply parse out the value for you using properties (not maintaining any additional state), and provide a way to easily create attribute definitions as well.

**Comment by mm21** [comment]
After thinking about this, I think a better solution is to write custom properties for accessing attributes, knowing ahead of time whether one or multiple values are expected. The following approach will work as of upcoming version 0.3.0:

```python
class MyNote(Note):
    
    @property
    def my_single_value_label(self) -> str:
        return self.labels.filter_name("mySingleValueLabel")[0].value

    @property
    def my_multi_value_label(self) -> list[str]:
        return [a.value for a in self.labels.filter_name("myMultiValueLabel")]
```

If you have a `Note` (e.g. from a search result) you can always create `MyNote` from its `note_id`:

```python
my_note = MyNote(note_id=note.note_id)
```
